backend/bin/kafkaConsumer/common/userCourseProgress/checkAndSendThresholdEmail.ts

```diff
@@ -0,0 +1,63 @@
+require("dotenv-safe").config({
+  allowEmptyValues: process.env.NODE_ENV === "production",
+})
+import { User, Course } from "@prisma/client"
+import { CombinedUserCourseProgress } from "./userFunctions"
+import { KafkaContext } from "../kafkaContext"
+import { v4 as uuidv4 } from "uuid"
+
+interface Props {
+  user: User
+  course: Course
+  combinedUserCourseProgress: CombinedUserCourseProgress
+  context: KafkaContext
+}
+
+export const checkAndSendThresholdEmail = async ({
+  user,
+  course,
+  combinedUserCourseProgress,
+  context,
+}: Props) => {
+  const courseEmailThresholdTemplates = await context.prisma.emailTemplate.findMany(
+    {
+      where: {
+        triggered_automatically_by_course_id: course.id,
+        template_type: "threshold",
+      },
+    },
+  )
+
+  const templatesThatFulfillPoints = courseEmailThresholdTemplates
+    .sort((a, b) => (b.points_threshold ?? 0) - (a.points_threshold ?? 0))
+    .filter((et) => {
+      return (
+        combinedUserCourseProgress.total_n_points >=
+        (et.points_threshold ?? 99999)
+      )
+    })
+
+  if (templatesThatFulfillPoints.length !== 0) {
+    // Send only for the highest fulfilled?
```

Comment:
I think the current design assumes other thresholds don't cancel other thresholds. If we wanted cancellability I think it should be an extra feature.

For now it's out of scope though.